examples/interacting_with_js_arrays/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+#[macro_use]
+extern crate neon;
+
+use neon::vm::{Call, JsResult};
+use neon::js::{Variant, JsNumber, JsArray};
+
+// https://github.com/dfrankland/node-eyeliner/blob/master/native/src/js_object_utils.rs
+pub fn js_value_to_float(js_number: Variant) -> f64 {
+  match js_number {
+    Variant::Number(number) => number.value(),
+    _ => panic!("heck"),
+  }
+}
+
+fn accepts_js_arrays(call: Call) -> JsResult<JsNumber> {
+  let js_arr_handle = call.arguments.get(call.scope, 0).unwrap();
+  let js_arr = js_arr_handle.downcast::<JsArray>().unwrap().to_vec(call.scope)?;
+
+  let vec_of_numbers: Vec<_> = js_arr.iter().map(move|js_value| {
```

Comment:
This is a great use case! Again let's try to avoid `Variant` and panics. The idiomatic Rust way to do this would be:

```rust
let vec_of_numbers: Vec<_> = js_arr.iter().map(|js_value| {
    let js_number = js_value::check::<JsNumber>()?;
    Ok(js_number.value())
}).collect()?;
```

Don't freak out if that seemed magical! It might be worth a comment. This is a good Rust idiom to learn but it's definitely a little dauntingly implicit at first. The short version of the story is, you can map with a function that returns a `Result` type and get a `Result<Vec<...>>` with nothing more than `.map(...).collect()`. Rust's iteration protocol is powerful enough to do the right thing.